### PR TITLE
Add tests for update with replacement

### PR DESCRIPTION
**Pull request by AlekSi**
## Readiness checklist

* [x] I added tests for new functionality or bugfixes.
* [x] I ran `task all`, and it passed.
* [x] I added/updated comments for both exported and unexported top-level declarations (functions, types, etc).
* [x] I checked comments rendering with `task godocs`.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

File: integration/setup/common.go [hunk 1]
```diff
@@ -54,10 +54,21 @@ var (
 //
 // This function should not be used lightly in new tests and should eventually be removed.
 func SkipForTigris(tb testing.TB) {
+	SkipForTigrisWithReason(tb, "")
+}
+
+// SkipForTigrisWithReason skips the current test for Tigris handler.
+//
+// This function should not be used lightly in new tests and should eventually be removed.
+func SkipForTigrisWithReason(tb testing.TB, reason string) {
```
Comment (rumyantseva) [inline integration/setup/common.go:63]: This is super handy! 👍

File: integration/update_field_compat_test.go [hunk 2]
```diff
@@ -55,14 +55,14 @@ func TestUpdateFieldCompatUnset(t *testing.T) {
 	testCases := map[string]updateCompatTestCase{
 		"Simple": {
 			update:        bson.D{{"$unset", bson.D{{"v", ""}}}},
-			skipForTigris: true,
+			skipForTigris: "TODO",
```
Comment (rumyantseva) [inline integration/update_field_compat_test.go:58]: Should we create an issue for these `TODO`s?

File: internal/handlers/tigris/msg_update.go [hunk 3]
```diff
@@ -210,7 +211,15 @@ func (h *Handler) update(ctx context.Context, sp fetchParam, doc *types.Document
 	h.L.Sugar().Debugf("Update: %s", u)
 
 	res, err := h.driver.UseDatabase(sp.db).Update(ctx, sp.collection, f, u)
-	if err != nil {
+	switch err := err.(type) {
+	case nil:
+		// do nothing
+	case *driver.Error:
+		if isInvalidArgument(err) {
```
Comment (rumyantseva) [inline internal/handlers/tigris/msg_update.go:218]: I moved such checks to `tigrisdb`, so who merges the second will need to move this one too :P

**Review by rumyantseva — CHANGES_REQUESTED** [review]
WLS linter and some tests fail, this is why I reject.

I don't see critical things, so when it's fixed, I will be ready to approve.

File: integration/update_field_compat_test.go (hunk 2 above)
Comment (AlekSi) [reply integration/update_field_compat_test.go:58]: Yes, but probably not right now

**Review by w84thesun — COMMENTED** [review]
Please resolve conflicts.

**Review by noisersup — DISMISSED** [review]
LGTM

File: internal/handlers/tigris/tigris.go [hunk 4]
```diff
@@ -78,10 +78,18 @@ func isNotFound(err *driver.Error) bool {
 	}
 
 	//nolint:nosnakecase // Tigris named their const that way
-	if err.Code == api.Code_NOT_FOUND {
-		return true
+	return err.Code == api.Code_NOT_FOUND
```
Comment (rumyantseva) [inline internal/handlers/tigris/tigris.go:81]: 👍

**Review by rumyantseva — COMMENTED** [review]
The tests still fail, right? Do you need any help with it?

File: internal/handlers/tigris/tigrisdb/errors.go [hunk 5]
```diff
@@ -48,3 +48,18 @@ func IsAlreadyExists(err *driver.Error) bool {
...
+
+// IsInvalidArgument returns true if the error is "invalid argument" error.
+// This function is implemented to keep nolint in a single place.
+func IsInvalidArgument(err *driver.Error) bool {
+	if err == nil {
+		return false
+	}
+
+	//nolint:nosnakecase // Tigris named their const that way
+	if err.Code == api.Code_INVALID_ARGUMENT {
```
Comment (rumyantseva) [inline internal/handlers/tigris/tigrisdb/errors.go:60]: I remember you had that `return err.Code == api.Code_INVALID_ARGUMENT`. Did you decide not to use it?

**Review by rumyantseva — APPROVED** [review]
Looks good to me, one tiny question.

File: internal/handlers/tigris/tigrisdb/errors.go (hunk 5 above)
Comment (AlekSi) [reply internal/handlers/tigris/tigrisdb/errors.go:60]: Well, maybe later. Chasing a different problem now :)

Comment (rumyantseva) [reply internal/handlers/tigris/tigrisdb/errors.go:60]: 👌

**Review by noisersup — CHANGES_REQUESTED** [review]
Tests are still failing

**Review by w84thesun — CHANGES_REQUESTED** [review]
I saw those PG `currentDate` tests failed several times. Should we create an issue for that?

**Comment by AlekSi** [comment]
> I saw those PG `currentDate` tests failed several times. Should we create an issue for that?

I will try to fix the underlying issue tomorrow

**Review by rumyantseva — APPROVED** [review]
If the tests pass, I'm happy with the changes.

**Review by w84thesun — APPROVED** [review]

**Comment by AlekSi** [comment]
@noisersup PTAL

**Review by noisersup — APPROVED** [review]

**Pull request merged**